kong/pdk/client.lua

```diff
@@ -241,6 +242,50 @@ local function new(self)
     return is_tls and "tls" or "tcp"
   end
 
+  ---
+  -- Returns a table of `authenticated_groups` of the currently authenticated
+  -- consumer.
+  -- @function kong.client.get_authenticated_groups
+  -- @phases access, header_filter, body_filter, log
+  -- @treturn table,nil|err
+  -- The table will have an array part to iterate over, and a hash part where each
+  -- group name is indexed by itself. Eg.
+  -- {
+  --   [1] = "users",
+  --   [2] = "admins",
+  --   users = "users",
+  --   admins = "admins",
+  -- }
+  -- @usage
+  -- local groups, err = kong.client.get_authenticated_groups()
+  -- if not err then
+  --   for i = 1, #groups_to_check do
+  --     if groups[i] == 'group1' then
+  --       return true
+  --     end
+  --   end
+  -- end
+  function _CLIENT.get_authenticated_groups()
+    check_phase(AUTH_AND_LATER)
+
+    local authenticated_groups = ngx.ctx.authenticated_groups
+    if authenticated_groups == nil then
+      return nil
+    end
+
+    if type(authenticated_groups) ~= "table" then
+      return nil, "invalid authenticated_groups, a table was expected"
+    end
```

Comment:
do we need to test for this? if it's not a table a hard error will occur, which is ok, since this is a issue that should surface with a stacktrace probably.